File: java/class/extends/Cover.java
public class Cover{
    public static void main(String[] args){
        People p1 = new People("人类", 23); 
        p1.say();
        Student stu1 = new Student("小花", 25, 98);
        stu1.say();
        stu1.say("哈哈");
    }
}

class People{
   String name;
   int age;
   People(String name, int age){
        this.name = name;
        this.age = age;
   }
   void say(){
        System.out.println("People, People name = "+name+", age="+age);
   }
}

class Student extends People{
    int score;
    Student(String name, int age, int score){
        super(name, age); 
        this.score = score;
    }
    void say(){
        System.out.println("Student, Student name = "+name+", age="+age); 
    }

    void say(String word){
        System.out.println("Student say "+word); 
    }
}

File: java/class/extends/Cover.cpp
#include <iostream>
using namespace std;
class People{
    public:
        const char *name;
        int age;
    public:
        People(const char *name, int age):name(name), age(age){
            cout<<"People name="<<name<<",age="<<age<<endl; 
        }
        void say(){
            cout<<"People say"<<endl; 
        }
};

class Student: public People{
    public:
        int score;
    public:
        Student(const char *name, int age, int score):People(name, age){
            this->score = score; 
            cout<<"Student name="<<name<<",age="<<age<<",score"<<score<<endl; 
        }

        void say(){
            cout<<"Student say"<<endl; 
        }

        void say(const char *word){
            cout<<"Student say"<<word<<endl; 
        }
};

int main(){
    People p1 = People("人类", 30);
    p1.say();

    Student stu1 = Student("小花", 30, 98);
    stu1.say();
    stu1.say("哈哈");

    //指针
    Student *stu2 = new Student("小明", 20, 70);
    stu2->say();
    delete stu2;

    return 0;
}
